.github/workflows/build-postgres.yml

```diff
@@ -0,0 +1,37 @@
+name: Build Postgres
+
+on:
+  workflow_call:
+  schedule:
+    - cron: "0 0 * * *"
+
+permissions:
+  id-token: write
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@b5ca514318bd6ebac0fb2aedd5d36ec1b5c232a2 # v3.10.0
+        with:
+          # We are overriding the default buildkit version being used by Buildx. We need buildkit >= 12.0 and currently BuildX
```

Comment:
This was copied from other workflows and kept intact for consistency.

Reply:
We should check whether we still need this pinning. At the time, we only needed it because the buildkit version in CI was too old but that is over a year ago now.

I don't remember what the original issue was but the Git history should tell us.

Reply:
Will open a separate PR for this.